<a href="https://colab.research.google.com/github/viandre7/Viviana_GarciaMonje_KC/blob/main/Taller_Semana_8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**1. TALLER SEMANA 8**



## **1.1 BREAST CANCER (CLASIFICASIÖN)**


### **1.1.1 Notación**

**1) ID number** (Se elimina)

**2) Diagnosis** (M = malignant, B = benign) 
Ten real-valued features are computed for each cell nucleus:

**a) radius** (mean of distances from center to points on the perimeter)

**b) texture** (standard deviation of gray-scale values)

**c) perimeter**

**d) area**

**e) smoothness** (local variation in radius lengths)

**f) compactness** (perimeter^2 / area - 1.0)

**g) concavity** (severity of concave portions of the contour)

**h) concave points** (number of concave portions of the contour)

**i) symmetry**

**j) fractal dimension** ("coastline approximation" - 1)

###**1.1.2 Conectarse al drive**


In [2]:
from google.colab import drive
import os
drive.mount('/content/gdrive')
# Establecer ruta de acceso en drive
import os
print(os.getcwd())
os.chdir("/content/gdrive/My Drive")

Mounted at /content/gdrive
/content


In [3]:
%cd '/content/gdrive/MyDrive/G2 - Data Sience 2021 (Keycode)/Taller_Semana_8'

/content/gdrive/MyDrive/G2 - Data Sience 2021 (Keycode)/Taller_Semana_8


###**1.1.3 Cargar paquetes**


In [43]:
#Carga de las librerías
import urllib.request
import pandas as pd

from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score



###**1.1.4 Cargar los datos - Breast_cancer (Clasificación)**


In [14]:
data_cancer= pd.read_csv('Breast_cancer_1.csv',sep=',')
data_cancer= data_cancer.drop(columns=['id'])
data_cancer.head()

diagnosis  radius_mean  ...  fractal_dimension_worst  Unnamed: 32
0         M        17.99  ...                  0.11890          NaN
1         M        20.57  ...                  0.08902          NaN
2         M        19.69  ...                  0.08758          NaN
3         M        11.42  ...                  0.17300          NaN
4         M        20.29  ...                  0.07678          NaN

[5 rows x 32 columns]

**Nota** 

Eliminamos la columna id, ya que no nos aporta en nada, sumado a que estos modelos no deben incluir dicha columna para su análisis

In [6]:
data_cancer.shape

(569, 32)

###**1.1.5 Remmplazar datos en diagnosis**

In [15]:
data_cancer['diagnosis1']=data_cancer.diagnosis.replace(to_replace=['B', 'M'], value=[0, 1])
data_cancer.head()

diagnosis  radius_mean  ...  Unnamed: 32  diagnosis1
0         M        17.99  ...          NaN           1
1         M        20.57  ...          NaN           1
2         M        19.69  ...          NaN           1
3         M        11.42  ...          NaN           1
4         M        20.29  ...          NaN           1

[5 rows x 33 columns]

In [16]:
data_cancer.drop(['diagnosis', 'Unnamed: 32'], axis = 'columns', inplace=True)

**Nota** 

Creamos una columna nueva. diagnosis1, para guardar los datos de la columna diagnosis que vamos a reemplazar (M,B) por 0 y 1, donde 0 es Benigno y 1 es Maligno, posteriormente eliminamos la colomna diagnosis ya que no la necesitamos, y la columna Unnamed pues no nos aporta nada.

In [17]:
data_cancer.head()

radius_mean  texture_mean  ...  fractal_dimension_worst  diagnosis1
0        17.99         10.38  ...                  0.11890           1
1        20.57         17.77  ...                  0.08902           1
2        19.69         21.25  ...                  0.08758           1
3        11.42         20.38  ...                  0.17300           1
4        20.29         14.34  ...                  0.07678           1

[5 rows x 31 columns]

In [ ]:
data_cancer.isnull().sum()

In [19]:
data_cancer.shape

(569, 31)

**NOTA**

El dataset se compone de 31 columnas de las cuales 30 corresponden a variables independientes, que usaremos para predecir.

diagnosis1 es la variable a predecir. Todas las variables son numéricas, ya sea enteras o reales y no tiene valores nulos.

###**1.1.6 Graficos**

###**1.1.7 Definir la matriz de diseño (X) y el vector respuesta (y)**

In [52]:
X = data_cancer.drop(["diagnosis1"], axis=1) # Tomar todas las columnas excepto la ultima
y = data_cancer["diagnosis1"] # Tomar el vector respuesta (ultima columna)

In [53]:
X

radius_mean  texture_mean  ...  symmetry_worst  fractal_dimension_worst
0          17.99         10.38  ...          0.4601                  0.11890
1          20.57         17.77  ...          0.2750                  0.08902
2          19.69         21.25  ...          0.3613                  0.08758
3          11.42         20.38  ...          0.6638                  0.17300
4          20.29         14.34  ...          0.2364                  0.07678
..           ...           ...  ...             ...                      ...
564        21.56         22.39  ...          0.2060                  0.07115
565        20.13         28.25  ...          0.2572                  0.06637
566        16.60         28.08  ...          0.2218                  0.07820
567        20.60         29.33  ...          0.4087                  0.12400
568         7.76         24.54  ...          0.2871                  0.07039

[569 rows x 30 columns]

In [54]:
y


0      1
1      1
2      1
3      1
4      1
      ..
564    1
565    1
566    1
567    1
568    0
Name: diagnosis1, Length: 569, dtype: int64

In [55]:
y.unique()

array([1, 0])

###**1.1.8 Separar en train y tests**

In [56]:
from sklearn.model_selection import train_test_split
(X_train, X_test,y_train, y_test) = train_test_split(X,y,stratify=y,test_size=0.30,random_state=42)

**Nota** Eligieremos el 70% de los datos para entrenamiento, y el 30% para test, de igual manera se le aplicara el random_state para seleccionar la semilla y los datos sean aleatoreos

###**1.1.9 Entrenar el modelo - Random Forest**

In [57]:
model = RandomForestClassifier(random_state=42, n_estimators=100,
class_weight="balanced", max_features="log2")
model.fit(X_train, y_train) #Se ajusta a los datos de entrenamiento

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight='balanced',
                       criterion='gini', max_depth=None, max_features='log2',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=42, verbose=0,
                       warm_start=False)

**Nota** 
Numero de estimadores 100 (arboles), tipo balanceado para que no queden muchas muestras separadas en la seleccion, tomara un log2 para tomar la cantidad de columnas para cada arbol

###**1.1.10 Sacar predicciones**

In [ ]:
# import numpy as np
# a=np.array(y_test)
# a

In [60]:
y_test_pred = model.predict(X_test) #Prediccion en Test
y_test_pred

array([0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
       1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0,
       0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1,
       1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1,
       0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1])

###**1.1.11 Calculamos el Acuracy**

In [61]:
# from sklearn.metrics import accuracy_score

test_accuracy = accuracy_score(y_test, y_test_pred)
print('% de aciertos sobre el set de evaluación:',test_accuracy)

% de aciertos sobre el set de evaluación: 0.9766081871345029


**Nota** obtenemos un acuracy de 0.97, es decir, una probabilidad de el 97% de los datos que predice sean correctos, de cada 100 datos que se clasifiquen el 97 quedaran bien clasificados, lo cual nos indica que es un muy buen modelo

###**1.1.12 Calculamos la precisión**

In [62]:
from sklearn.metrics import precision_score
precision_score(y_test, y_test_pred) 

1.0

**Nota**  El porcentaje de muestras que se clasifican bien dentro de cada clase, es aproximadamente del %

###**1.1.13 Matriz de confusion**

In [63]:
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test, y_test_pred) 

array([[107,   0],
       [  4,  60]])

El modelo tuvo 107 verdaderos positivos, 0 falsos positivos, 60 verdaderos negativos y 4 falsos negativos, por lo que se puede concluir que el modelo funciona relativamente bien en cuanto a su clasificación

###**1.1.14 Calculamos el Recall**

In [64]:
from sklearn.metrics import recall_score
recall_score(y_test, y_test_pred) 

0.9375

**Nota** El modelo se adapta en un 93% a los datos al momento de clasificar

###**1.1.15 Calculamos f1 score**

In [65]:
from sklearn.metrics import f1_score
f1_score(y_test, y_test_pred) 

0.967741935483871

###**1.1.16 Conclusiones**

Todos los datos son similares, estan cerca, lo cual nos dice que el modelo es balanceado.

Con el random forest, obtenemos un acuracy de 0.97, es decir, el 97% de los datos que predice es correcto, lo cual nos indica que es un muy buen modelo

## **1.2 WINE (REGRESION)**


In [45]:
import pandas                  as pd
from   scipy import stats
import numpy                   as np
import matplotlib.pyplot       as plt
import seaborn                 as sns
import statsmodels.formula.api as sm
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


### **1.2.1 Notación**

1) Alcohol

2) Malic acid

3) Ash

4) Alcalinity of ash

5) Magnesium

6) Total phenols

7) Flavanoids

8) Nonflavanoid phenols

9) Proanthocyanins

10)Color intensity

11)Hue

12)OD280/OD315 of diluted wines

13)Proline

### **1.2.2 Cargar los datos**

In [35]:
data_wine= pd.read_csv('wine_1.csv',sep=',')
data_wine= data_wine.drop(columns=['Wine'])
data_wine.head()

Alcohol  Malic.acid   Ash   Acl  ...  Color.int   Hue    OD  Proline
0    14.23        1.71  2.43  15.6  ...       5.64  1.04  3.92     1065
1    13.20        1.78  2.14  11.2  ...       4.38  1.05  3.40     1050
2    13.16        2.36  2.67  18.6  ...       5.68  1.03  3.17     1185
3    14.37        1.95  2.50  16.8  ...       7.80  0.86  3.45     1480
4    13.24        2.59  2.87  21.0  ...       4.32  1.04  2.93      735

[5 rows x 13 columns]

Nota

Eliminamos la columna wine, la cual contiene valores de 1,2,3, se considera que clasifica a los vinos en la categoria 1,2 o 3.

In [36]:
data_wine.shape

(178, 13)

In [37]:
data_wine.isnull().sum()

Alcohol                 0
Malic.acid              0
Ash                     0
Acl                     0
Mg                      0
Phenols                 0
Flavanoids              0
Nonflavanoid.phenols    0
Proanth                 0
Color.int               0
Hue                     0
OD                      0
Proline                 0
dtype: int64

**NOTA**

El dataset se compone de 13 columnas de las cuales 12 corresponden a variables independientes, que usaremos para predecir.

Proline es la variable a predecir. Todas las variables son numéricas, ya sea enteras o reales y no tiene valores nulos.

### **1.2.3 Definir la matriz de diseño (X) y el vector respuesta (y)**

In [38]:
X=data_wine.iloc[:,:-1].values # Tomar todas las columnas excepto la ultima
y= data_wine.iloc[:,-1].values # Tomar el vector respuesta (ultima columna)

In [39]:
print(X.shape)
print(y.shape)

(178, 12)
(178,)


In [40]:
X

array([[14.23,  1.71,  2.43, ...,  5.64,  1.04,  3.92],
       [13.2 ,  1.78,  2.14, ...,  4.38,  1.05,  3.4 ],
       [13.16,  2.36,  2.67, ...,  5.68,  1.03,  3.17],
       ...,
       [13.27,  4.28,  2.26, ..., 10.2 ,  0.59,  1.56],
       [13.17,  2.59,  2.37, ...,  9.3 ,  0.6 ,  1.62],
       [14.13,  4.1 ,  2.74, ...,  9.2 ,  0.61,  1.6 ]])

### **1.2.4 Normalizamos los datos, usando Standard scaler**

Las variables quedarán entre -2 y 2

In [41]:
from sklearn.preprocessing import StandardScaler
X_n= StandardScaler().fit_transform(X)
X_n

array([[ 1.51861254, -0.5622498 ,  0.23205254, ...,  0.25171685,
         0.36217728,  1.84791957],
       [ 0.24628963, -0.49941338, -0.82799632, ..., -0.29332133,
         0.40605066,  1.1134493 ],
       [ 0.19687903,  0.02123125,  1.10933436, ...,  0.26901965,
         0.31830389,  0.78858745],
       ...,
       [ 0.33275817,  1.74474449, -0.38935541, ...,  2.22423597,
        -1.61212515, -1.48544548],
       [ 0.20923168,  0.22769377,  0.01273209, ...,  1.83492299,
        -1.56825176, -1.40069891],
       [ 1.39508604,  1.58316512,  1.36520822, ...,  1.79166599,
        -1.52437837, -1.42894777]])

In [46]:
y_1= np.transpose(np.array([y])) # Llevar a un vector vertical porque estaba como un vector horizontal
y_1.shape

(178, 1)

In [47]:
y_n= StandardScaler().fit_transform(y_1) # Estandarizacion al vector y
y_n

array([[ 1.01300893],
       [ 0.96524152],
       [ 1.39514818],
       [ 2.33457383],
       [-0.03787401],
       [ 2.23903902],
       [ 1.72952002],
       [ 1.74544249],
       [ 0.94931905],
       [ 0.94931905],
       [ 2.43010864],
       [ 1.69767508],
       [ 1.82505483],
       [ 1.28369089],
       [ 2.54793491],
       [ 1.79320989],
       [ 1.69767508],
       [ 1.22000102],
       [ 2.97147258],
       [ 0.3124203 ],
       [ 0.10542821],
       [ 0.07358327],
       [ 0.91747411],
       [ 0.85378424],
       [ 0.3124203 ],
       [ 0.2646529 ],
       [ 1.42699311],
       [ 1.71359755],
       [ 0.53533487],
       [ 0.91747411],
       [ 1.71359755],
       [ 2.44603111],
       [ 0.7741719 ],
       [ 1.55437286],
       [ 1.10854374],
       [ 0.55125733],
       [ 0.42387759],
       [ 1.14038868],
       [ 0.86970671],
       [ 0.04173834],
       [ 0.15319562],
       [ 0.91747411],
       [ 1.10854374],
       [-0.21302116],
       [ 0.43980005],
       [ 1

In [48]:
print('Shape de X',X_n.shape)
print('Shape de y',y_n.shape)

Shape de X (178, 12)
Shape de y (178, 1)


### **1.2.5 Separamos en entranamiento y test**

In [49]:
X_train, X_test, y_train, y_test = train_test_split(X_n, y_n, test_size=0.25, random_state=42,shuffle=True)

**Nota** Eligieremos el 70% de los datos para entrenamiento, y el 30% para test, de igual manera se le aplicara el random_state para seleccionar la semilla y los datos sean aleatoreos

### **1.2.6 Entrenamos el modelo**

In [50]:
regressor = LinearRegression() #Creamos el modelo
regressor.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

### **1.2.7 Analizamos los coeficientes**

In [51]:
#Recuperamos la intersección
print(regressor.intercept_)

[-0.00115046]


Valor mínimo de Proline


In [52]:
#Pendiente para cada una de las variables
print(regressor.coef_)

[[ 0.30658283 -0.14064782  0.21899154 -0.29325455  0.10389822  0.09177859
  -0.07767957 -0.01656427  0.07151682  0.22910699  0.07454604  0.13955565]]


Por cada unidad que aumente el Proline, las variables aumentarán en los valores de coef_ respectivamente

In [53]:
regressor.coef_[0]

array([ 0.30658283, -0.14064782,  0.21899154, -0.29325455,  0.10389822,
        0.09177859, -0.07767957, -0.01656427,  0.07151682,  0.22910699,
        0.07454604,  0.13955565])

In [55]:
data_wine_x= data_wine.drop(columns=['Proline'])
data_wine_x.columns

Index(['Alcohol', 'Malic.acid', 'Ash', 'Acl', 'Mg', 'Phenols', 'Flavanoids',
       'Nonflavanoid.phenols', 'Proanth', 'Color.int', 'Hue', 'OD'],
      dtype='object')

### **1.2.5 Calculamos los vectores de media y desviacion estandar**

In [58]:
np.mean(data_wine_x,axis=0)

Alcohol                 13.000618
Malic.acid               2.336348
Ash                      2.366517
Acl                     19.494944
Mg                      99.741573
Phenols                  2.295112
Flavanoids               2.029270
Nonflavanoid.phenols     0.361854
Proanth                  1.590899
Color.int                5.058090
Hue                      0.957449
OD                       2.611685
dtype: float64

In [109]:
np.std(data_wine_x,axis=0)

Wine                     0.772855
Alcohol                  0.809543
Malic.acid               1.114004
Ash                      0.273572
Acl                      3.330170
Mg                      14.242308
Phenols                  0.624091
Flavanoids               0.996049
Nonflavanoid.phenols     0.124103
Proanth                  0.570749
Color.int                2.311765
Hue                      0.227929
OD                       0.707993
dtype: float64

In [26]:
df=pd.DataFrame(regressor.coef_[0], data_wine_x.columns, columns=['Coeficiente'])
df['Medias']=np.mean(data_wine_x,axis=0)
df['Desviacion']=np.std(data_wine_x,axis=0)
df['Coef_final']=df.Medias + (df.Coeficiente*df.Desviacion) # Calculo del coeficiente sin normalizar
df

Coeficiente     Medias  Desviacion  Coef_final
Wine                    -0.759907   1.938202    0.772855    1.350904
Alcohol                  0.152939  13.000618    0.809543   13.124429
Malic.acid              -0.073977   2.336348    1.114004    2.253938
Ash                      0.122225   2.366517    0.273572    2.399954
Acl                     -0.085450  19.494944    3.330170   19.210381
Mg                       0.080930  99.741573   14.242308  100.894203
Phenols                  0.157384   2.295112    0.624091    2.393334
Flavanoids              -0.421339   2.029270    0.996049    1.609596
Nonflavanoid.phenols    -0.049123   0.361854    0.124103    0.355758
Proanth                  0.079403   1.590899    0.570749    1.636218
Color.int                0.367415   5.058090    2.311765    5.907467
Hue                      0.034409   0.957449    0.227929    0.965292
OD                      -0.073746   2.611685    0.707993    2.559474

### **1.2.5 Calculamos las predicciones**

In [66]:
#Hacemos nuestras predicciones
y_pred = regressor.predict(X_test)
y_pred

array([[ 0.96453059],
       [ 0.56485563],
       [-0.26634055],
       [ 0.66815138],
       [-0.51286369],
       [ 1.12668714],
       [-1.13709417],
       [-0.32404416],
       [-0.21448974],
       [-0.22037489],
       [ 0.52530428],
       [-0.08241104],
       [-0.28885415],
       [-0.85914041],
       [ 1.24745993],
       [-0.87334379],
       [-0.72018677],
       [-0.82192459],
       [ 1.14775087],
       [-0.67129189],
       [ 1.48831665],
       [ 0.21148955],
       [-0.26503957],
       [-0.0362105 ],
       [-0.25511601],
       [-0.03456092],
       [-0.7127638 ],
       [ 0.16662538],
       [-0.73881336],
       [ 0.76861764],
       [ 0.89838431],
       [-1.02071839],
       [-0.83845149],
       [ 0.71413963],
       [ 0.8218205 ],
       [ 1.23091432],
       [ 1.13720536],
       [-1.16572167],
       [-0.18138266],
       [ 0.69680575],
       [ 0.16113572],
       [-0.63783551],
       [ 0.01026801],
       [ 0.21155155],
       [-0.43569663]])

In [60]:
from sklearn import metrics 
print('Mean Absolute Error:', metrics.mean_absolute_error(y_test, y_pred))
print('Mean Squared Error:', metrics.mean_squared_error(y_test, y_pred))
print('Root Mean Squared Error:', np.sqrt(metrics.mean_squared_error(y_test, y_pred)))

Mean Absolute Error: 0.5500206423138361
Mean Squared Error: 0.470064620174384
Root Mean Squared Error: 0.6856125875262092


In [61]:
media_y= np.mean(data_wine.Proline)
desviacion_y= np.mean(data_wine.Proline)
desviacion_y

746.8932584269663

In [62]:
df_n= pd.DataFrame(X_test, columns=data_wine_x.columns)
df_n['y_pred']=y_pred
df_n['y_pred_final']=(y_pred*desviacion_y)+media_y
df_n

Alcohol  Malic.acid       Ash  ...        OD    y_pred  y_pred_final
0   0.789806    0.685502  0.707247  ...  1.056952  0.964531   1467.294651
1   1.493907    1.529305  0.268606  ...  1.014578  0.564856   1168.780118
2  -0.087232    0.425180  1.218995  ... -0.426113 -0.266341    547.965300
3   0.900980   -0.750759  1.218995  ...  0.138864  0.668151   1245.931021
4  -0.778980   -1.046988 -1.632171  ...  1.226445 -0.512864    363.838823
5   1.605081   -0.373741  1.292101  ...  0.054117  1.126687   1588.408288
6  -1.236028    0.981731 -1.339744  ...  0.619094 -1.137094   -102.394708
7   0.493343    1.412609  0.414820  ... -1.485445 -0.324044    504.866857
8  -1.717782   -0.885409  1.218995  ...  0.915707 -0.214490    586.692316
9   0.443932    0.200764 -0.060375  ... -0.200123 -0.220375    582.296736
10  0.616869   -0.472483  0.890014  ...  1.706675  0.525304   1139.239487
11  0.616869    0.703455  0.926567  ... -1.852681 -0.082411    685.341012
12  0.505695    1.349773 -0.901103  ...  0.548472 -0.288854    531.150038
13 -0.284874    0.981731 -1.412851  ... -0.694478 -0.859140    105.207076
14  0.777454   -0.472483  1.218995  ...  0.378979  1.247460   1678.612669
15 -0.593691    0.084068 -0.718336  ...  0.237735 -0.873344     94.598667
16 -1.964835   -1.432983  0.487926  ... -0.426113 -0.720187    208.990618
17 -1.137207   -1.082894  0.524480  ...  0.152988 -0.821925    133.003320
18  1.061565   -0.885409 -0.352802  ...  1.325316  1.147751   1604.140645
19 -1.137207   -0.849502  0.487926  ...  0.816836 -0.671292    245.509870
20  1.469202   -0.669969  0.414820  ...  0.294232  1.488317   1858.506928
21  0.135116   -1.190614 -2.436346  ...  0.802712  0.211490    904.853379
22 -0.828391   -1.109824 -0.316249  ... -1.330077 -0.265040    548.936987
23  0.493343    2.031997  1.803849  ... -0.976966 -0.036211    719.847878
24 -0.284874    0.048161 -0.316249  ... -1.386574 -0.255116    556.348828
25  0.962743    0.380297 -0.243142  ... -1.315952 -0.034561    721.079944
26 -0.717217   -0.652016 -0.645229  ...  0.491974 -0.712764    214.534785
27 -0.778980   -1.011081  0.707247  ... -0.440238  0.166625    871.344630
28 -1.137207   -0.454530 -0.170035  ... -0.482611 -0.738813    195.078541
29  0.690985   -0.544297  0.341713  ...  0.590845  0.768618   1320.968595
30  1.259207   -0.589180 -0.572122  ...  1.381813  0.898384   1417.890441
31 -0.778980   -0.634063 -0.243142  ...  0.237735 -1.020718    -15.474425
32  0.196879    1.107404 -0.791443  ... -1.315952 -0.838451    120.659492
33  0.715690   -0.607133 -0.023821  ...  0.378979  0.714140   1280.279333
34  0.925685   -0.544297  0.158946  ...  0.407228  0.821820   1360.705448
35  1.086270   -0.400670  0.816907  ...  1.339440  1.230914   1666.254863
36  1.654492   -0.589180  1.218995  ... -0.920468  1.137205   1596.264275
37 -0.581338    2.848870  0.999674  ... -1.301828 -1.165722   -123.776401
38 -0.778980   -1.136754 -0.974210  ...  0.223610 -0.181383    611.419771
39  0.592164   -0.598156  0.999674  ... -1.174708  0.696806   1267.332774
40  0.085705   -0.750759 -0.974210  ...  0.110615  0.161136    867.244439
41 -0.964270   -0.939268 -1.559065  ...  0.576721 -0.637836    270.498214
42 -0.408401   -1.217544 -0.462462  ...  0.774463  0.010268    754.562369
43  0.419227   -1.253450 -0.023821  ... -0.962842  0.211552    904.899684
44  0.765101    2.346179 -0.060375  ... -0.793348 -0.435697    421.474384

[45 rows x 14 columns]